# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: caps/nsIPrincipal.idl
# Commit: 39f2e21623aa
# Full Hash: 39f2e21623aa7c19517e32cbd97dc1c3584c58ca
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2020-06-02 03:24:46
# Regressor Bug: 1638358
# File Overlap Count: 1
# Description:
#   Bug 1638358 - Cookie Schemeful Same-Site - part 5 - schemeful comparison, r=mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D75629
# ==============================================================================

diff -r 2c873c72bf1f -r 39f2e21623aa caps/nsIPrincipal.idl
--- a/caps/nsIPrincipal.idl	Mon Jun 01 16:49:05 2020 +0000
+++ b/caps/nsIPrincipal.idl	Mon Jun 01 16:49:03 2020 +0000
@@ -287,7 +287,12 @@
     /* Returns the Spec of the Principals URI with
     * user/pass/ref/query stripped for privacy and spoof prevention
     */
-     readonly attribute ACString exposableSpec;
+    readonly attribute ACString exposableSpec;
+
+    /**
+     * Return the scheme of the principals URI
+     */
+    readonly attribute ACString scheme;
 
     /**
     * Checks if the Principal's URI Scheme matches with the parameter